src/django_components/tag_formatter.py

```diff
@@ -0,0 +1,269 @@
+
+        This function receives a list of tokens
+
+        `['component', '"my_comp"', 'key=val', 'key2=val2']`
+
+        `component` is the tag name, which we drop. `"my_comp"` is the component name,
+        but we must remove the extra quotes. And we pass remaining tokens unmodified,
+        as that's the input to the component.
+
+        So in the end, we return a tuple:
+
+        `('my_comp', ['key=val', 'key2=val2'])`
+        """
+        ...
+
+    @abc.abstractmethod
+    def parse_inline_tag(self, tokens: List[str]) -> Tuple[str, List[str]]:
+        """
+        Given the tokens (words) of a component inline tag, this function extracts
+        the component name from the tokens list, and returns a tuple of
+        `(component_name, component_input)`.
+
+        Example:
+
+        Given a component declarations:
+
+        `{% #my_comp key=val key2=val2 %}`
+
+        This function receives a list of tokens
+
+        `['#my_comp', 'key=val', 'key2=val2']`
+
+        We drop the leading hash from `#my_comp` to obtain the component name.
+        And we pass remaining tokens unmodified, as that's the input to the component.
+
+        So in the end, we return a tuple:
+
+        `('my_comp', ['key=val', 'key2=val2'])`
+        """
+        ...
+
+    # NOTE: We validate the generated tags, so they contain only valid characters (\w - : . @ #)
+    # and NO SPACE. Otherwise we wouldn't be able to distinguish a "multi-word" tag from several
+    # single-word tags.
+    def validate_tag(self, tag: str, tag_type: str) -> None:
+        if not tag:
+            raise ValueError(
+                f"{self.__class__.__name__} returned an invalid tag for {tag_type}: '{tag}'." f" Tag cannot be empty"
+            )
+
+        if not TAG_RE.match(tag):
+            raise ValueError(
+                f"{self.__class__.__name__} returned an invalid tag for {tag_type}: '{tag}'."
+                f" Tag must contain only following chars: {VAR_CHARS}"
+            )
+
+    def safe_format_block_start_tag(self, name: str) -> str:
+        """Formats the start tag of a block component. Raises ValueError if the tag is invalid."""
+        tag = self.format_block_start_tag(name)
+        self.validate_tag(tag, "block_start_tag")
+        return tag
+
+    def safe_format_block_end_tag(self, name: str) -> str:
+        """Formats the end tag of a block component. Raises ValueError if the tag is invalid."""
+        tag = self.format_block_end_tag(name)
+        self.validate_tag(tag, "block_end_tag")
+        return tag
+
+    def safe_format_inline_tag(self, name: str) -> str:
+        """Formats the start tag of an inline component. Raises ValueError if the tag is invalid."""
+        tag = self.format_inline_tag(name)
+        self.validate_tag(tag, "inline_tag")
+        return tag
+
+
+# TODO - Update docs that these are names of positional args
+# TODO - Add tests
+class ComponentTagFormatter(TagFormatterABC):
```

Comment:
This is the default tag formatter, which has the same behavior as current. Except this also adds an "inlined" tag `#component`.